Automattic/wp-calypso
# Add warning for mapped domain with wrong NS config

This fixes #3152.

Goes with D2003-code.

There's a bug where mapped domain notices don't work, fix in #6983 and D2348-code.

This adds a warning notice when you buy a mapped domain but don't update the nameserver records to point to our name servers.

There are three place where this appears:

1. My site sidebar
2. Domain Management -> Domain List
3. Domain Management -> Domain Details

@ranh or @breezyskies:  I had to change GridIcon size from 16 to 18 (It wouldn't let me commit), fyi. 

Screenshot:
-
![image](https://cloud.githubusercontent.com/assets/991022/13063857/5cb42c38-d401-11e5-899c-4ee77b5e41f1.png)


Testing:
-

- Get a new mapped domain
- Confirm you see the warning on 3 places aforementioned.
- Configure your NS records to point to WordPress name servers
- Confirm the message is gone or you can compare it with an already configured domain.


PS: Although we'd like the message to visible to make people configure their site, there are some legitimate (albeit not endorsed by us) ways to get a mapped domain working without changing NS records. So there's a slight chance that seeing this message everywhere might be annoying for some.

/cc: @klimeryk @aidvu for code
/cc: @ranh @breezyskies for copy and ideas on ^
